Sources/SpeziStudyDefinition/StudyBundle.swift

```diff
@@ -0,0 +1,319 @@
+                rawValue = try container.decode(String.self)
+            }
+            
+            public func encode(to encoder: any Encoder) throws {
+                var container = encoder.singleValueContainer()
+                try container.encode(rawValue)
+            }
+            
+            public static let consent = Self(rawValue: "consent")
+            public static let questionnaire = Self(rawValue: "questionnaire")
+            public static let informationalArticle = Self(rawValue: "informationalArticle")
+        }
+        
+        public let category: Category
+        public let filename: String
+        public let fileExtension: String
+        
+        public var description: String {
+            "\(category.rawValue)/\(filename).\(fileExtension)"
+        }
+        
+        public init(category: Category, filename: String, fileExtension: String) {
+            self.category = category
+            self.filename = filename
+            self.fileExtension = fileExtension
+        }
+    }
+    
+    
+    public struct LocalizationKey: Hashable, LosslessStringConvertible, Sendable {
+        public let language: Locale.Language
+        public let region: Locale.Region
+        
+        public var description: String {
+            language.minimalIdentifier + "-" + region.identifier
+        }
+        
+        public init(locale: Locale) {
+            self.init(language: locale.language, region: locale.region!)
+        }
+        
+        public init(language: Locale.Language, region: Locale.Region) {
+            self.language = language
+            self.region = region
+        }
+        
+        public init?(_ description: String) {
+            let components = description.split(separator: "-")
+            guard components.count == 2 else {
+                return nil
+            }
+            let languageIdentifier = components[0]
+            let regionIdentifier = components[1]
+            guard let language = Locale.Language.systemLanguages.first(where: { $0.minimalIdentifier == languageIdentifier }),
+                  let region = Locale.Region.isoRegions.first(where: { $0.identifier == regionIdentifier }) else {
+                return nil
+            }
+            self.init(language: language, region: region)
+        }
+        
+        func score(against locale: Locale) -> Double {
+            let languageMatches = locale.language.minimalIdentifier == self.language.minimalIdentifier
+            let regionMatches = locale.region?.identifier == self.region.identifier // TODO also allow mayching against parent regions? (eg: if the user is in DE, but the region in the key is EU, that should match...)
+            switch (languageMatches, regionMatches) {
+            case (true, true):
+                return 1 // perfect score
+            case (true, false): // correct language, wrong region
+                return 0
```

Comment:
Shouldn't this be ranked a bit higher compared to the fact that language and region both won't match?

Reply:
i think it might be best to allow externally specifying/overriding the selection rules here (with a sensible default as a fallback).

i still feel like the region should rank higher than the language, but in this particular case i just haven't changed it yet to your preferred behavior

Reply:
I think it is common to rather rank language higher than region in case only one of them fits. Region is most of the time only a small variation of language. I see that we can make this configurable in the future but might not be needed at the very beginning.

Reply:
changed. we now have a `LocaleMatchingBehaviour` enum with a couple of differen cases (`requirePerfectMatch`, `preferRegionOverLanguage`, `preferLanguageOverRegion`, `custom`) and with a `default` option, which is `preferRegionOverLanguage`